resources/post-upgrade.sh

```diff
@@ -137,6 +148,138 @@ migrateServiceAccounts() {
+
+      # Now replace the random secret in the generated JSON with your extracted secret
+      if [ -n "$secret" ]; then
+        # Assume the created file name format is <app>-<id>.json
+        json_output_file=$(ls -1 "$SERVICE_REGISTRY_PRODUCTION"/${app}-*.json | head -n 1)  # Get the most recently created file
+
+        if [[ -f "$json_output_file" ]]; then
+          # Assuming the random secret is in the generated file, replace it with secret from migration
+          sed -i "s|\"clientSecret\": \".*\"|\"clientSecret\": \"$secret\"|" "$json_output_file"
+        else
+          echo "Error: JSON file for $app not found."
+        fi
+      fi
+
+      echo "Service configuration for $app created."
+  done
+
+  doguctl config "service_accounts/migrated" "true"
+
+  echo "Migration completed. Individual files created for each application."
+}
+
+migrateLegacyServicesFromETCD() {
+  basePath="/tmp/legacyEtcdCasMigration"
+
+  mkdir -p "${basePath}"
+
+  # Get all services listed in etcd
+  wget -O "${basePath}/ces.json" "http://$(getEtcdEndpoint):4001/v2/keys/dogu_v2?recursive=true"
+
+  # extract the keys from json etcd response
+  jq '.node.nodes' "${basePath}/ces.json" > "${basePath}/nodes.json"
+  # filter services that have a dependency on cas
+  jq -r '.[] | .nodes[] | select(.value | contains("\"name\":\"cas\"")) | .value' "${basePath}/nodes.json" | jq -s '.' > "${basePath}/filter.json"
+  # filter services that don't have defined a service account for cas within dogu.json
+  jq '[.[] | select((.ServiceAccounts == null) or (.ServiceAccounts? | map(select(.Type == "cas")) | length) == 0)]' "${basePath}/filter.json" > "${basePath}/exclude.json"
+  # extract name from dogu.json
+  jq -r '.[] | .Name ' "${basePath}/exclude.json" > "${basePath}/migration.json"
+  # delete namespace from name, sort and delete duplicates
+  sed 's#.*/##' "${basePath}/migration.json" | sort | uniq > "${basePath}/migrationCandidates.txt"
+
+  cat "${basePath}/migrationCandidates.txt"
+
+  candidateFile="${basePath}/migrationCandidates.txt"
+
+  # Check if names file exists
+  if [[ ! -f "$candidateFile" ]]; then
+      echo "candidate file not found: $candidateFile" >&2
+      exit 1
+  fi
+
+  # Read each line from the file
+  while IFS= read -r name; do
+      echo "checking ${name}..."
+      #Check whether service is installed and has not migrated already
+      status_code=$(wget --spider -S "http://$(getEtcdEndpoint):4001/v2/keys/dogu_v2/${name}/current" 2>&1 | grep "HTTP/" | awk '{print $2}'; exit 0)
+      echo "status code from wget is: ${status_code}"
+      if [[ "$status_code" -eq 200 ]]; then
+        # We do not need to consider the LogoutUri as it has already been migrated in a previous step.
+        doguctl config "service_accounts/cas/${name}/created" "true"
+        echo "set service account entry in etcd for service ${name}"
+      fi
+  done < "$candidateFile"
+
+  rm -r "${basePath}"
+
+  echo "Legacy service account migration done."
```

Comment:
```suggestion
  echo "Legacy service account migration done.\n"
```

to separate output in console